src/projectviewerwindow.cpp

```diff
@@ -0,0 +1,80 @@
+#include "projectviewerwindow.h"
+#include "ui_projectviewerwindow.h"
+
+#include <QDebug>
+#include <QPushButton>
+#include <QStringList>
+#include <QStringListModel>
+#include <QItemSelection>
+#include <QItemSelectionModel>
+#include <QStandardItemModel>
+
+
+
+ProjectViewerWindow::ProjectViewerWindow(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::ProjectViewerWindow)
+{
+    ui->setupUi(this);
+
+    ui->projectsListView->setViewMode(QListView::ViewMode::IconMode);
+    ui->projectsListView->setIconSize(QSize(100,100));
+    ui->projectsListView->setUniformItemSizes(true);
+    ui->projectsListView->setSelectionMode(QAbstractItemView::MultiSelection);
+
+    connect(ui->newProjectButton,&QPushButton::clicked,this,&ProjectViewerWindow::onNewProjectButtonClicked);
+    connect(ui->openProjectButton,&QPushButton::clicked,this,&ProjectViewerWindow::onOpenProjectButtonClicked);
+    connect(ui->deleteProjectButton,&QPushButton::clicked,this,&ProjectViewerWindow::onRemoveProjectButtonClicked);
+
+}
+ProjectViewerWindow::~ProjectViewerWindow()
+{
+    delete ui;
+}
+
+void ProjectViewerWindow::setModel(QAbstractItemModel *model)
+{
+    if(model)
+    {
+        mProjectsViewModel = model;
+        ui->projectsListView->setModel(mProjectsViewModel);
+        mSelectionModel = ui->projectsListView->selectionModel();
+        connect(mSelectionModel,&QItemSelectionModel::selectionChanged,this,&ProjectViewerWindow::makeButtonsEnabled);
+    }
+}
+
+void ProjectViewerWindow::onNewProjectButtonClicked()
+{
+    emit newProjectRequested();
+    qDebug()<<"new project button clicked";
+}
+
+void ProjectViewerWindow::onOpenProjectButtonClicked()
+{
+    emit projectOpened("");
+}
+
+void ProjectViewerWindow::onRemoveProjectButtonClicked()
+{
+    emit projectRemoved("");
+}
+
+void ProjectViewerWindow::makeButtonsEnabled()
+{
+    qDebug()<<mSelectionModel->selectedIndexes().size();
+    if(mSelectionModel->selectedIndexes().size()== 1)
+    {
```

Comment:
You do not need so many code to do the same. Open project is enabled if selection count == 1, delete button enabled if selection count > 0. That is 2 bool values that could be set right away.

Reply:
However, I still get 2 conditions and one else block.
Like this. Or I'm a fool)
![Screenshot from 2019-06-30 20-46-08](https://user-images.githubusercontent.com/32616149/60400110-24f61a80-9b78-11e9-9575-e52bb303ab48.png)

Reply:
Could you please assign each condition to separate bool variable. And everything will become clear.

Reply:
Done)